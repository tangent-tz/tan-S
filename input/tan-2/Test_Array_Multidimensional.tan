main {
    var a := [1.1, 2.2, 3, 4, 5]; 
    print a \n [a:1] \n "hello!" \n; 
    
    var ints := [[[1, 2, 3], [4, 5]], [[3, 4], [7, 8, 9]], [[5, 6], [10, 12]]]; 
    print ints \n; 
    
    print [[[1, 2, 3], [4, 5]], [[3, 4], [7, 8, 9]], [[5, 6], [10, 12]]] \n\n; 
    
    #-----------------------------------------------------------------------------
    
    const floats := [[[[1.1, 1.2, 1.3], [1.4, 1.5, 1.6]]], [[[1.7, 1.8], [1.9, 2.1]]]];
    print floats \n; 
    
    print [[[[1.1, 1.2, 1.3], [1.4, 1.5, 1.6]]], [[[1.7, 1.8], [1.9, 2.1]]]] \n\n;
    
    #-----------------------------------------------------------------------------
    
    const strings := [[[["hello", "tanzil", "sarker"], ["it's", "me", "ivy :)"]]], [[["how", "r"], ["u", "??"]]]];
    print strings \n; 
    
    print [[[["hello", "tanzil", "sarker"], ["it's", "me", "ivy :)"]]], [[["how", "r"], ["u", "??"]]]] \n\n;

    #-----------------------------------------------------------------------------
    
    const chars := [['a','b','c','d'],['e', %146]]; 
    print chars \n;
    
    print [['a','b','c','d'],['e', %146]] \n\n; 
    
    #-----------------------------------------------------------------------------
    
    const bools := [[[true, true], [false, false]], [[true, false], [false, true]]];
    print bools \n; 
    
    print [[[true, true], [false, false]], [[true, false], [false, true]]] \n\n; 
    
    
    
    
    #-----------------------------------------------------------------------------
    print \n "LOOP TEST 1:" \n; 
    var x := 0; 
    while (x < 10) {
        var y := [1, 2, 3]; 
        [y:2] := x; 
        print y \n; 
        print y \n; 
        x := x + 1; 
    }
    
    
    
    print \n "LOOP TEST 2:" \n;
    const loopInts := [[[[4, 5, 6]], [[7, 8, 9]]], [[[13, 14, 15]], [[22, 23, 24]]], [[[30, 31]]]]; 
    x := 0; 
    while (x < length loopInts) {
        var outer := [loopInts:x]; 
        print outer \n; 
        
        var j := 0; 
        while (j < length outer) {
            print [outer:j] \n; 
            j := j + 1; 
        }
        print \n; 
        x := x + 1; 
    }
    
    
    
    #---------------------------------------------------------------------------------------
    # TANZIL'S TEST
    {   
        print \n "Tanzil's test:" \n; 
        var boolean := [true,false];
        print boolean \n;
        
        var multiDimBoolean := [[true,false],[false,true]];
        print multiDimBoolean \n;
        
        var integer := [1,2,3];
        print integer \n;
        
        var multiDimInteger := [[1,2,3],[4,5,6]];
        print multiDimInteger \n;
        
        var floatarray := [1.1,2.2,3.3];
        print floatarray \n;
        
        var multiDimFloat := [[1.1,2.2,3.3],[4.4,5.5,6.6]];
        print multiDimFloat \n;
        
        var stringarray := ["hrty67"];
        print stringarray \n;
        
        
        var multiDimString := [["hello","world"],["goodbye","world"]];
        print multiDimString \n;
        
        var multiNestedsameType := [[[[4,5,6]],[[7,8,9]]],[[[13,14,15]],[[22,23,24]]]];
        print multiNestedsameType \n;
        
        var multiNestedFloatType := [[[[[4.5],[5.5],[6.5]]],[[[7.5],[8.5],[9.5]]]],[[[[13.5],[14.5],[15.5]]],[[[22.5],[23.5],[24.5]]]]];
        print multiNestedFloatType \n;
        
        var x := 0;
        while (x < 10) {
            var y := [1,2,3];
            [y:2] := x;
            print y \n;
            x := x + 1;
        }
        
        
        var test := [1];
        print test \n;
        
        var test2 := [[[1]]];
        print test2 \n;
        
        var test3 := [[[1]]];
        
        print test3 \n;
    }
 
}