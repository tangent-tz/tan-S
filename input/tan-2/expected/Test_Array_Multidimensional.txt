[ 1.100000, 2.200000, 3.000000, 4.000000, 5.000000 ]
2.200000
hello!
[ [ [ 1, 2, 3 ], [ 4, 5 ] ], [ [ 3, 4 ], [ 7, 8, 9 ] ], [ [ 5, 6 ], [ 10, 12 ] ] ]
[ [ [ 1, 2, 3 ], [ 4, 5 ] ], [ [ 3, 4 ], [ 7, 8, 9 ] ], [ [ 5, 6 ], [ 10, 12 ] ] ]

[ [ [ [ 1.100000, 1.200000, 1.300000 ], [ 1.400000, 1.500000, 1.600000 ] ] ], [ [ [ 1.700000, 1.800000 ], [ 1.900000, 2.100000 ] ] ] ]
[ [ [ [ 1.100000, 1.200000, 1.300000 ], [ 1.400000, 1.500000, 1.600000 ] ] ], [ [ [ 1.700000, 1.800000 ], [ 1.900000, 2.100000 ] ] ] ]

[ [ [ [ hello, tanzil, sarker ], [ it's, me, ivy :) ] ] ], [ [ [ how, r ], [ u, ?? ] ] ] ]
[ [ [ [ hello, tanzil, sarker ], [ it's, me, ivy :) ] ] ], [ [ [ how, r ], [ u, ?? ] ] ] ]

[ [ a, b, c, d ], [ e, f ] ]
[ [ a, b, c, d ], [ e, f ] ]

[ [ [ true, true ], [ false, false ] ], [ [ true, false ], [ false, true ] ] ]
[ [ [ true, true ], [ false, false ] ], [ [ true, false ], [ false, true ] ] ]


LOOP TEST 1:
[ 1, 2, 0 ]
[ 1, 2, 0 ]
[ 1, 2, 1 ]
[ 1, 2, 1 ]
[ 1, 2, 2 ]
[ 1, 2, 2 ]
[ 1, 2, 3 ]
[ 1, 2, 3 ]
[ 1, 2, 4 ]
[ 1, 2, 4 ]
[ 1, 2, 5 ]
[ 1, 2, 5 ]
[ 1, 2, 6 ]
[ 1, 2, 6 ]
[ 1, 2, 7 ]
[ 1, 2, 7 ]
[ 1, 2, 8 ]
[ 1, 2, 8 ]
[ 1, 2, 9 ]
[ 1, 2, 9 ]

LOOP TEST 2:
[ [ [ 4, 5, 6 ] ], [ [ 7, 8, 9 ] ] ]
[ [ 4, 5, 6 ] ]
[ [ 7, 8, 9 ] ]

[ [ [ 13, 14, 15 ] ], [ [ 22, 23, 24 ] ] ]
[ [ 13, 14, 15 ] ]
[ [ 22, 23, 24 ] ]

[ [ [ 30, 31 ] ] ]
[ [ 30, 31 ] ]


Tanzil's test:
[ true, false ]
[ [ true, false ], [ false, true ] ]
[ 1, 2, 3 ]
[ [ 1, 2, 3 ], [ 4, 5, 6 ] ]
[ 1.100000, 2.200000, 3.300000 ]
[ [ 1.100000, 2.200000, 3.300000 ], [ 4.400000, 5.500000, 6.600000 ] ]
[ hrty67 ]
[ [ hello, world ], [ goodbye, world ] ]
[ [ [ [ 4, 5, 6 ] ], [ [ 7, 8, 9 ] ] ], [ [ [ 13, 14, 15 ] ], [ [ 22, 23, 24 ] ] ] ]
[ [ [ [ [ 4.500000 ], [ 5.500000 ], [ 6.500000 ] ] ], [ [ [ 7.500000 ], [ 8.500000 ], [ 9.500000 ] ] ] ], [ [ [ [ 13.500000 ], [ 14.500000 ], [ 15.500000 ] ] ], [ [ [ 22.500000 ], [ 23.500000 ], [ 24.500000 ] ] ] ] ]
[ 1, 2, 0 ]
[ 1, 2, 1 ]
[ 1, 2, 2 ]
[ 1, 2, 3 ]
[ 1, 2, 4 ]
[ 1, 2, 5 ]
[ 1, 2, 6 ]
[ 1, 2, 7 ]
[ 1, 2, 8 ]
[ 1, 2, 9 ]
[ 1 ]
[ [ [ 1 ] ] ]
[ [ [ 1 ] ] ]
